docs/blog/posts/argocd-keptn-health.md

```diff
@@ -0,0 +1,163 @@
+---
+date: 2024-05-15
+authors: [odubajDT]
+description: >
+  In this blog post we will present how Keptn can show application health in integration with ArgoCD.
+categories:
+  - ArgoCD
+  - Application health
+comments: true
+---
+
+# Application health checks with Keptn using ArgoCD
+
+In this blog post we will present a planned Keptn and
+[ArgoCD](https://argo-cd.readthedocs.io/en/stable/) integration to execute
+advanced application health checks using Keptn, where the application itself is deployed
+by ArgoCD and display the Keptn application health status via ArgoCD UI.
+
+Keptn provides an effective way to perform application health checks using the
+pre- or post-deployment tasks and evaluations.
+Compared to ArgoCD application
+health checks, which are evaluating if the application is successfully deployed
+and the workloads are running on the cluster, they do not show if the microservices
+of a single application are able to communicate and cooperate with each other.
+Keptn pre- and post-deployment tasks and evaluations complement the missing functionality
+by providing a straight-forward way to examine the application ability to perform
+actions for which it was developed to do.
+
+<!-- more -->
+
+## How it's going to work?
+
+The integration of Keptn and ArgoCD performing and displaying full application
+health checks is not fully implemented yet, but
+we can have a brief look how it might work in the future.
+
+First of all, Keptn and ArgoCD need to be installed and enabled on the same
+cluster.
+The reason is, we want to have ArgoCD performing the actual deployment
+of the application and Keptn executing the advanced application health checks.
+
+Additionally, we will need to have an ArgoCD extension, which will consist of
+React application extending the ArgoCD UI, implemented as
+[ArgoCD UI Application Tab Extension](https://argo-cd.readthedocs.io/en/stable/developer-guide/extensions/ui-extensions/#application-tab-extensions)
+and a proxy allowing Keptn (which will work as a backend service)
+to push the application health status data to the ArgoCD UI.
+For proxy setup, the
+[ArgoCD proxy extension](https://argo-cd.readthedocs.io/en/stable/developer-guide/extensions/proxy-extensions/)
+will be used.
```

Comment:
This is a bit too specific without getting into too many details 🤔 
I would rather expand and explain these in detail or remove the paragraph.

Reply:
I think it's ok, it gives the reader a brief overview of what is going to be done. ArgoCD has it documented pretty well.